jquery.overlay.js

```diff
@@ -211,23 +216,40 @@
           // Style attribute's string
           style = 'background-color:' + strategy.css['background-color'];
 
-          text.contents().each(function () {
-            var text, html, str, prevIndex;
-            if (this.nodeType != Node.TEXT_NODE) return;
-            text = this.textContent;
+          if (this.allowOverlapping) {
+            var textContent, prevIndex, str, html;
+            textContent = text.html()
             html = '';
             for (prevIndex = match.lastIndex = 0;; prevIndex = match.lastIndex) {
-              str = match.exec(text);
+              str = match.exec(textContent);
               if (!str) {
-                if (prevIndex) html += escape(text.substr(prevIndex));
+                if (prevIndex) html += textContent.substr(prevIndex);
                 break;
               }
               str = str[0];
-              html += escape(text.substr(prevIndex, match.lastIndex - prevIndex - str.length));
-              html += '<span style="' + style + '">' + escape(str) + '</span>';
+              html += textContent.substr(prevIndex, match.lastIndex - prevIndex - str.length);
+              html += '<span style="' + style + '">' + str + '</span>';
             };
-            if (prevIndex) $(this).replaceWith(html);
-          });
+            if (prevIndex) text.html(html);
+          } else {
+            text.contents().each(function () {
+              var text, html, str, prevIndex;
+              if (this.nodeType != Node.TEXT_NODE) return;
+              text = this.textContent;
+              html = '';
+              for (prevIndex = match.lastIndex = 0; ; prevIndex = match.lastIndex) {
```

Comment:
I'm ok with it, but if there's any reasonably low effort way to prevent infinite loops, I think we should put in the stopgap. The problem with an infinite loop like this is that it *might* break in such a way that javascript never gets to call for help, and if it breaks (say) every 250th time somebody uses it, it might take a long time before somebody catches it and takes the time to file a support ticket. 

Also, definitely document the reasons that this is safe so that somebody doesn't change them!